File: 백준/Silver/1026. 보물/보물.js
const filePath =
  process.platform === "linux" ? "/dev/stdin" : "1026/example.txt";
const input = require("fs")
  .readFileSync(filePath)
  .toString()
  .trim()
  .split("\n");

let N = parseInt(input.shift());

const A = input
  .shift()
  .split(" ")
  .map(Number)
  .sort((a, b) => b - a);
const B = input
  .shift()
  .split(" ")
  .map(Number)
  .sort((a, b) => a - b);

const result = A.reduce((prev, cur, index) => {
  return prev + parseInt(cur) * parseInt(B[index]);
}, 0);

console.log(result);

File: 백준/Silver/1026. 보물/보물.py
N = int(input())

A = list(map(int, input().split()))
B = list(map(int, input().split()))

A = sorted(A)
# B = sorted(B, reverse=True)
# B = sorted(B, key=lambda x:-x)
B = sorted(B)[::-1] # split으로 내림차순 하는 방법

ans = 0
for i in range(N):
    ans += A[i] * B[i]

print(ans)